enterprise/internal/batches/store/worker_workspace_execution.go

```
@@ -197,6 +197,7 @@ func (s *batchSpecWorkspaceExecutionWorkerStore) MarkComplete(ctx context.Contex
 	if err != nil {
 		return false, err
 	}
+	defer func() { err = tx.Done(err) }()
```

There is some funkiness happening in this method. I think the change is safe, but it seems strange to me that `rollbackAndMarkFailed` will not return the error that caused the rollback, and will instead just return `nil`. It seems like that will rollback, but swallow the error, making it look to the caller like everything worked as expected.